tools/docker/tests/test_flows.sh

```diff
@@ -0,0 +1,106 @@
+#find prplmesh folder
+scriptdir="$(cd "${0%/*}"; pwd)"
+installdir="${scriptdir%/*/*/*/*}/build/install"
+
+dbg() {
+    [ "$VERBOSE" = "true" ] && echo "$@"
+}
+
+status() {
+	printf '\033[1;35m'"$@\n"'\033[0m'
+}
+
+err() {
+    printf '\033[1;31m'"$@\n"'\033[0m'
+}
+
+success() {
+    printf '\033[1;32m'"$@\n"'\033[0m'
+}
+
+run() {
+    dbg "$*"
+    "$@" || exit $?
+}
+
+report() {
+    msg="$1"; shift
+    if "$@"; then
+        success "OK $msg"
+    else
+        err "FAIL $msg"
+        error=1
+    fi
+}
+
+start_gw_repeater() {
+    dbg "delete running containers before starting test"
+    docker rm -f $(docker ps -a -q) >/dev/null 2>&1
+    dbg "start gw and repeater"
+    ${scriptdir}/test_gw_repeater.sh >/dev/null 2>&1
+}
+
+send_topology_query() {
+    dbg "Sending toplogy query to al_mac $1"
+    docker exec -it gateway ${installdir}/bin/beerocks_cli -c "bml_trigger_topology_discovery $1" >/dev/null 2>&1
+}
+
+check_log_for_query() {
+    dbg "Confirming topology query was received"
+    docker exec -it repeater cat tmp/$USER/beerocks/logs/beerocks_agent.log | grep -i -q "Topology Query"
+}
+
+test_topology() {
+    status "test topology query"
+    send_topology_query $1
+    check_log_for_query
```

Comment:
I don't think it makes much sense to have functions for send_topology_query and check_for_log_query - they will be used in only one place, and they're very simple.

Conversely, it might be useful to have functions for triggering a BML command:
```
send_bml_command() {
    docker exec -it gateway ${installdir}/bin/beerocks_cli -c "$*"
}
```
(note the `"$*"`, not the usual `"$@"`, because beerock_cli supports only one argument to -c).

The rest can just as well be inlined here directly I think.

(nitpick)